### DISCUSSION TITLE
Directly include Base64-encoded images in .qmd

### DISCUSSION THREAD

**Discussion by zeileis** [quarto-dev/quarto-cli#6567] [Q&A]
### Description

When I include Base64-encoded images directly in .qmd files, they are not rendered correctly. When I just do

```
![](data:image/png;base64,i [...snip...] =)
```

Then a .png suffix is added at the end in the `<img =src"..">` which does not work.

Additionally, in some contexts I get the `file name too long (os error 36)` error, similar to https://github.com/quarto-dev/quarto-cli/issues/4613.

A minimal example replicating the first issue using quarto 1.3.450 and R package [base64enc](https://CRAN.R-project.org/package=base64enc) is:

````qmd
Output of `image(volcano)` in Base64:
```{r}
#| echo: false
#| results: "asis"
## create PNG file
img <- tempfile(fileext = ".png")
png(img)
image(volcano)
dev.off()

## write as base 64
img64 <- sprintf("![](%s)", base64enc::dataURI(file = img, mime = "image/png"))
writeLines(img64)
```
````

I wasn't able to pinpoint exactly when I get the `file name too long` error. I only get it with Opengraph/Twitter enabled in `_quarto.yml` but only in my actual website and not in a minimal fresh website. I can try to dig deeper in case this is needed.

**Comment by cscheid** [comment]
I see Pandoc is definitely appending a `.png` to the end of our data URIs, and that is 100% a bug we need to fix. I'll open an issue.

**Comment by mcanouil** [comment]
Closing in favour of:
- https://github.com/quarto-dev/quarto-cli/issues/6568
